### ISSUE TITLE
Pirates Bay

### ISSUE THREAD

**Issue by jriker1** [Jackett/Jackett#3022]
Current Jackett in Windows.

When I try to download a torrent thru Sonarr using Jackett, I get the below from Pirates Bay:

2018-04-23 20:13:23.4940 Error Error downloading thepiratebay https://proxytpb.org/torrent/21168736/General_Hospital_-S56_E14[14039]_-_2018-04-19 BencodeNET.Exceptions.InvalidBencodeException`1[BencodeNET.Objects.IBObject]: Failed to parse IBObject. Invalid beginning character of object. Found ‘<’ at position 0. Valid characters are: 0-9, ‘i’, ‘l’ and 'd’
at BencodeNET.Parsing.BencodeParser.Parse(BencodeStream stream)
 at BencodeNET.Parsing.BencodeParser.Parse(Byte[] bytes)
 at Jackett.Controllers.DownloadController.d__5.MoveNext() in C:\projects\jackett\src\Jackett\Controllers\DownloadController.cs:line 75

The torrent downloads fine direct using the magnetic link but not thru Sonarr linked to Jackett.  Note in Sonarr I get the below not found messages which isn't actually the case as I mentioned direct I can download fine.  I think there is a larger core issue with things as searches are returning limited results now also but here's what Sonarr says:

18-4-23 20:13:15.9|Info|DownloadDecisionMaker|Processing 2 releases
 18-4-23 20:13:23.4|Warn|HttpClient|HTTP Error - Res: [GET] http://127.0.0.1:9117/dl/thepiratebay/?jackett_apikey=(removed)&path=UUkrVGlJZmsxaStrcWwzNXV2RC9rR21JbkZ5YXloVkFPanVXU3Y0c1QvaGUyWnB6bU1jbnhqemRHcDE3UlBSRzk0T3R2S2xlWWFSSFFQQVBtdlpXSDUrUmJaRWs0VEpvUHNzQlVjNm9laXFDMVBZSnNTLzFwTUNrdHhrbXFNSDNlM0dpMDRKbmtMb2xTQXRRMHlWdzAzZkJtUk5VeGlCVDFmL2UzZzkyNzh4Q2kxTnlzK01TWENKRmJ0YXBpcDhxdmtYdTZ6cGhLRk52RVZIak9oVENleWROczlhSFR3QnQ3S0VaTnFocHBnVXU2UG5Gc2V4alVTRkRURCtaTTA5Y1Z5b0FoSEpmOGhYRm5qYVZTaTU1TEg4QURNaDNKTWswNHVZUE5BZmlZZEs1UDhYTzlRM3p2Qk1DVnFCV3c1dWR3d0x5NXFIejRvcXp1a0dOcER5d0paZW5xVDhWKzJxTHZPanJiL21WRlcyenpKcUpHMG9Udy9XTzU5cC95OGFLb1NpdGp1bzlzM01GcStPbnJoRXdNWGkzU2w0U2pnWDlpYllZOVV1TmFSTT01&file=General+Hospital+-+S56+E14+[14039]+-+2018-04-19: 404.NotFound
 18-4-23 20:13:23.4|Error|Transmission|Downloading torrent file for episode ‘General Hospital - S56 E14 [14039] - 2018-04-19’ failed since it no longer exists (http://127.0.0.1:9117/dl/thepiratebay/?jackett_apikey=(removed)&path=UUkrVGlJZmsxaStrcWwzNXV2RC9rR21JbkZ5YXloVkFPanVXU3Y0c1QvaGUyWnB6bU1jbnhqemRHcDE3UlBSRzk0T3R2S2xlWWFSSFFQQVBtdlpXSDUrUmJaRWs0VEpvUHNzQlVjNm9laXFDMVBZSnNTLzFwTUNrdHhrbXFNSDNlM0dpMDRKbmtMb2xTQXRRMHlWdzAzZkJtUk5VeGlCVDFmL2UzZzkyNzh4Q2kxTnlzK01TWENKRmJ0YXBpcDhxdmtYdTZ6cGhLRk52RVZIak9oVENleWROczlhSFR3QnQ3S0VaTnFocHBnVXU2UG5Gc2V4alVTRkRURCtaTTA5Y1Z5b0FoSEpmOGhYRm5qYVZTaTU1TEg4QURNaDNKTWswNHVZUE5BZmlZZEs1UDhYTzlRM3p2Qk1DVnFCV3c1dWR3d0x5NXFIejRvcXp1a0dOcER5d0paZW5xVDhWKzJxTHZPanJiL21WRlcyenpKcUpHMG9Udy9XTzU5cC95OGFLb1NpdGp1bzlzM01GcStPbnJoRXdNWGkzU2w0U2pnWDlpYllZOVV1TmFSTT01&file=General+Hospital+-+S56+E14+[14039]+-+2018-04-19)

[v2.0.0.5163] NzbDrone.Common.Http.HttpException: HTTP request failed: [404:NotFound] [GET] at [http://127.0.0.1:9117/dl/thepiratebay/?jackett_apikey=(removed)&path=UUkrVGlJZmsxaStrcWwzNXV2RC9rR21JbkZ5YXloVkFPanVXU3Y0c1QvaGUyWnB6bU1jbnhqemRHcDE3UlBSRzk0T3R2S2xlWWFSSFFQQVBtdlpXSDUrUmJaRWs0VEpvUHNzQlVjNm9laXFDMVBZSnNTLzFwTUNrdHhrbXFNSDNlM0dpMDRKbmtMb2xTQXRRMHlWdzAzZkJtUk5VeGlCVDFmL2UzZzkyNzh4Q2kxTnlzK01TWENKRmJ0YXBpcDhxdmtYdTZ6cGhLRk52RVZIak9oVENleWROczlhSFR3QnQ3S0VaTnFocHBnVXU2UG5Gc2V4alVTRkRURCtaTTA5Y1Z5b0FoSEpmOGhYRm5qYVZTaTU1TEg4QURNaDNKTWswNHVZUE5BZmlZZEs1UDhYTzlRM3p2Qk1DVnFCV3c1dWR3d0x5NXFIejRvcXp1a0dOcER5d0paZW5xVDhWKzJxTHZPanJiL21WRlcyenpKcUpHMG9Udy9XTzU5cC95OGFLb1NpdGp1bzlzM01GcStPbnJoRXdNWGkzU2w0U2pnWDlpYllZOVV1TmFSTT01&file=General+Hospital+-+S56+E14+[14039]+-+2018-04-19]
 at NzbDrone.Common.Http.HttpClient.Execute(HttpRequest request) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Common\Http\HttpClient.cs:line 94
 at NzbDrone.Common.Http.HttpClient.Get(HttpRequest request) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Common\Http\HttpClient.cs:line 220
 at NzbDrone.Core.Download.TorrentClientBase`1.DownloadFromWebUrl(RemoteEpisode remoteEpisode, String torrentUrl) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Core\Download\TorrentClientBase.cs:line 134

18-4-23 20:13:23.4|Error|ReleaseModule|Downloading torrent failed

[v2.0.0.5163] NzbDrone.Core.Exceptions.ReleaseUnavailableException: Downloading torrent failed —> NzbDrone.Common.Http.HttpException: HTTP request failed: [404:NotFound] [GET] at [http://127.0.0.1:9117/dl/thepiratebay/?jackett_apikey=(removed)&path=UUkrVGlJZmsxaStrcWwzNXV2RC9rR21JbkZ5YXloVkFPanVXU3Y0c1QvaGUyWnB6bU1jbnhqemRHcDE3UlBSRzk0T3R2S2xlWWFSSFFQQVBtdlpXSDUrUmJaRWs0VEpvUHNzQlVjNm9laXFDMVBZSnNTLzFwTUNrdHhrbXFNSDNlM0dpMDRKbmtMb2xTQXRRMHlWdzAzZkJtUk5VeGlCVDFmL2UzZzkyNzh4Q2kxTnlzK01TWENKRmJ0YXBpcDhxdmtYdTZ6cGhLRk52RVZIak9oVENleWROczlhSFR3QnQ3S0VaTnFocHBnVXU2UG5Gc2V4alVTRkRURCtaTTA5Y1Z5b0FoSEpmOGhYRm5qYVZTaTU1TEg4QURNaDNKTWswNHVZUE5BZmlZZEs1UDhYTzlRM3p2Qk1DVnFCV3c1dWR3d0x5NXFIejRvcXp1a0dOcER5d0paZW5xVDhWKzJxTHZPanJiL21WRlcyenpKcUpHMG9Udy9XTzU5cC95OGFLb1NpdGp1bzlzM01GcStPbnJoRXdNWGkzU2w0U2pnWDlpYllZOVV1TmFSTT01&file=General+Hospital+-+S56+E14+[14039]+-+2018-04-19]
 at NzbDrone.Common.Http.HttpClient.Execute(HttpRequest request) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Common\Http\HttpClient.cs:line 94
 at NzbDrone.Common.Http.HttpClient.Get(HttpRequest request) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Common\Http\HttpClient.cs:line 220
 at NzbDrone.Core.Download.TorrentClientBase1.DownloadFromWebUrl(RemoteEpisode remoteEpisode, String torrentUrl) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Core\Download\TorrentClientBase.cs:line 134 --- End of inner exception stack trace --- at NzbDrone.Core.Download.TorrentClientBase1.DownloadFromWebUrl(RemoteEpisode remoteEpisode, String torrentUrl) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Core\Download\TorrentClientBase.cs:line 166
 at NzbDrone.Core.Download.TorrentClientBase`1.Download(RemoteEpisode remoteEpisode) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Core\Download\TorrentClientBase.cs:line 117
 at NzbDrone.Core.Download.DownloadService.DownloadReport(RemoteEpisode remoteEpisode) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Core\Download\DownloadService.cs:line 76
 at NzbDrone.Api.Indexers.ReleaseModule.DownloadRelease(ReleaseResource release) in M:\BuildAgent\work\b69c1fe19bfc2c38\src\NzbDrone.Api\Indexers\ReleaseModule.cs:line 67

Thanks.

JR

**Comment by garfield69** [comment]
I don't have any issues using TPB with Sonarr via Jackett.

But then again, I am using the official TPB site https://thepiratebay.org/, where as you are using an unofficial proxy, which btw, is flagged by both chrome and my anti-malware as a phishing site.

The problem you are running into, is that the magnet link the proxy supplies is not a *direct* magnet link, but a redirect to advertising and other links (hence the phishing flags), which only return the actual magnet link after you have clicked though several of these redirects.

Bottom line is, you can use this proxy with Jackett for searching, but not downloading.
If you are really want to be able to use TPB with Sonarr via Jackett, use the official TPB site, or find an unofficial proxy that provides direct magnet links in the same way that the official site does.
